components/core/src/clp_s/clp-s.cpp

```
@@ -116,6 +128,195 @@
+    } else {
+        out_path = option.irs_dir + "/" + path.substr(index, path.length() - index) + ".ir";
+    } */
+    std::filesystem::path input_path{path};
+    std::string filename = input_path.filename().string();
+    std::string out_path = option.irs_dir + "/" + filename + ".ir";
+
+    clp_s::FileWriter out_file;
+    out_file.open(out_path, clp_s::FileWriter::OpenMode::CreateForWriting);
+    clp_s::ZstdCompressor zc;
+    zc.open(out_file, option.compression_level);
+
```

cppcheck

[error] Exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)